File: src/test/java/com/pablisco/tests/MockitoExtensions.java
package com.pablisco.tests;

import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.withSettings;

public class MockitoExtensions {

    public static <T> T stub(Class<T> type) {
        return mock(type, withSettings().stubOnly());
    }

}

File: src/test/kotlin/com/pablisco/tests/MockitoExtensions.kt
package com.pablisco.tests

import org.mockito.BDDMockito.*

inline fun <reified T> mock(noinline f: T.() -> Unit = {}): T =
    mock(T::class.java).apply(f)

inline fun <reified T> stub(noinline f: T.() -> Unit = {}): T =
    mock(T::class.java, withSettings().stubOnly()).apply(f)

infix fun <T> T.willReturn(t: T) {
    given(this).willReturn(t)
}

infix fun <T> T.willThrow(ef: () -> Throwable) {
    given(this).willThrow(ef())
}
